pyquil/api/compiler.py

```diff
@@ -209,33 +212,41 @@ def apply_clifford_to_pauli(self, clifford, pauli_in):
             pauli_out *= PauliTerm(pauli, all_qubits[i])
         return pauli_out * pauli_in.coefficient
 
-    def _rb_sequence_payload(self, depth, qubits, gateset):
+    def _rb_sequence_payload(self, depth, gateset):
         """
-        Prepares a JSON payload for generating a randomized benchmarking sequence - see generate_rb_sequence.
+        Prepares a JSON payload for generating a randomized benchmarking sequence.
+
+        See :py:func:`generate_rb_sequence`.
 
         :param int depth: The number of cliffords per rb sequences to generate.
-        :param int qubits: The number of qubits to perform rb on.
-        :param list gateset: A list of Gate objects that make up the gateset to decompose the Cliffords into.
+        :param list gateset: A list of Gate objects that make up the gateset to decompose
+            the Cliffords into.
         :return: The JSON payload, with keys "depth", "qubits", and "gateset".
         """
+        # Support QubitPlaceholders: we index to arbitrary integers. `generate_rb_sequence`
+        # must handle mapping back to the original gateset gates.
```

Comment:
Do you have an idea of how to this cleanly? 

Currently this means that `generate_rb_sequence` is going to return programs that have been arbitrarily indexed, right?
At the very least the docstring for `generate_rb_sequence` should be updated to say this, but I think that if you give `generate_rb_sequence` programs with `QubitPlaceholder` objects then it should return programs with `QubitPlaceholder`, i.e. we should reconstruct the inverse mapping.

Reply:
`generate_rb_sequence` *does* reconstruct the inverse mapping (with no modification)

Reply:
I'll update this code comment to describe it better

Reply:
Oh my, you're right, NEAT.